onnxruntime/core/framework/bfc_arena.h

```diff
@@ -273,11 +291,24 @@ class BFCArena : public IArenaAllocator {
     RegionManager() = default;
     ~RegionManager() = default;
 
-    void AddAllocationRegion(void* ptr, size_t memory_size) {
+    void AddAllocationRegion(void* ptr, size_t memory_size, int64_t id) {
       // Insert sorted by end_ptr
       auto entry =
           std::upper_bound(regions_.begin(), regions_.end(), ptr, &Comparator);
-      regions_.insert(entry, AllocationRegion(ptr, memory_size));
+      regions_.insert(entry, AllocationRegion(ptr, memory_size, id));
+    }
+
+    Status RemoveAllocationRegion(void* ptr) {
+      auto entry =
+          std::upper_bound(regions_.begin(), regions_.end(), ptr, &Comparator);
+
+      if (entry == regions_.end()) {
+        LOGS_DEFAULT(FATAL) << "Could not find Region for " << ptr;
```

Comment:
fatal is catastrophic. would ERROR do?

Reply:
Actually I feel FATAL logging level is right here - it is catastrophic if this check fails -it means something is messed up in the arena book-keeping. The level of logging is consistent with the one in `RegionFor()` below.